Makefile

```diff
@@ -57,10 +61,36 @@ test: generate $(ALL_GO_MODS:%=test/%)
 test/%/go.mod:
 	@cd $* && $(GOCMD) test $(ARGS) ./...
 
-.PHONY: generate
-generate: export CFLAGS := $(BPF_INCLUDE)
-generate: go-mod-tidy
-generate:
+PROBE_ROOT = internal/pkg/instrumentation/bpf/
+PROBE_DIRS = $(shell find $(PROBE_ROOT) -type d -exec sh -c '(ls -p "{}"|grep />/dev/null)||dirname "{}"' \;)
+PROBE_GEN := $(addsuffix /bpf_x86_bpfel.go,$(PROBE_DIRS))
+PROBE_GEN += $(addsuffix /bpf_arm64_bpfel.go,$(PROBE_DIRS))
+PROBE_GEN := $(wildcard $(PROBE_GEN))
+PROBE_GEN += internal/pkg/instrumentation/bpf/net/http/client/bpf_no_tp_arm64_bpfel.go
+PROBE_GEN += internal/pkg/instrumentation/bpf/net/http/client/bpf_no_tp_x86_bpfel.go
+
+# Include all depinfo files to ensure we only re-generate when needed.
+-include $(shell find $(PROBE_ROOT) -type f -name 'bpf_*_bpfel.go.d')
+
+$(PROBE_GEN): %_bpfel.go: %_bpfel.o
+
+%/bpf_x86_bpfel.o: generate/%
+	$(info building $@)
+
+%/bpf_arm64_bpfel.o: generate/%
+	$(info building $@)
+
+%/bpf_no_tp_x86_bpfel.o: generate/%
+	$(info building $@)
+
+%/bpf_no_tp_arm64_bpfel.o: generate/%
+	$(info building $@)
+
+.PHONY: generate generate/all
+generate: go-mod-tidy $(PROBE_GEN)
+generate/%:
+	$(GOCMD) generate ./$*...
+generate/all:
```

Comment:
Do we want to call `generate/all` in CI and Dockerfile? If I understand this correctly, as long as there is no caching it should not make a difference. But maybe in the CI we want to make sure we generate all the files?

Reply:
I think if we checked these objects in to git that would be a good idea, but without that, I don't see a need.

FWIW, I've been reconsidering if we should be checking these into git. But that is a topic we can discuss outside of this PR.